package.json

```
@@ -170,9 +170,19 @@
             "default": false
           },
           "calva.enableClojureLspOnStart": {
-            "type": "boolean",
-            "markdownDescription": "**Recommended**: Start [clojure-lsp](https://calva.io/clojure-lsp/) automatically when Calva starts",
-            "default": true
+            "type": "string",
+            "enum": [
+              "when-workspace-opened",
+              "when-file-opened",
+              "never"
+            ],
+            "enumDescriptions": [
+              "Start clojure-lsp automatically when a workspace containing a Clojure project is opened",
+              "Start clojure-lsp automatically when a Clojure file is opened",
+              "Never start clojure-lsp automatically"
+            ],
+            "markdownDescription": "**Recommended**: Start [clojure-lsp](https://calva.io/clojure-lsp/) automatically when a Clojure file in a project is opened",
```

I think we can help with choosing the best option a bit more:

```suggestion
              "Start clojure-lsp automatically when a workspace containing a Clojure project is opened. Best option for single-Clojure-project workspaces and Polylith-like projects.",
              "Start clojure-lsp automatically when a Clojure file is opened.",
              "Don't start clojure-lsp automatically. You can always start clojure-lsp at will via the  command palette or from the statusbar. Good option in multi-project workspaces where you want to start a clojure-lsp for a particular project. Also useful if you are running on a machine with limited resources, like a Raspberry Pi."
            ],
            "markdownDescription": "Determines if [clojure-lsp](https://calva.io/clojure-lsp/) should start automatically, and what triggers it.",
```